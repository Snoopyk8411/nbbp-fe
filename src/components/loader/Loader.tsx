import cn from 'classnames';

import loaderStyles from './loader.module.css';

export const Loader = (): JSX.Element => <div className={cn(loaderStyles.loader, loaderStyles.theme)}></div>;
