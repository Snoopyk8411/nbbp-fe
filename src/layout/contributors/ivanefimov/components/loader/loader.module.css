.loader {
  --clr-accent-400: hsl(142, 90%, 61%);
  display: inline-block;
  width: 80px;
  height: 80px;
}
.loader:after {
  content: ' ';
  display: block;
  width: 64px;
  height: 64px;
  margin: 8px;
  border-radius: 50%;
  border: 6px solid var(--clr-accent-400);
  border-color: var(--clr-accent-400) transparent var(--clr-accent-400) transparent;
  animation: loader 1.2s linear infinite;
}
@keyframes loader {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
