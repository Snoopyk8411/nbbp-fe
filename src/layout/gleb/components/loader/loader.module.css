.loader {
    display: inline-block;
    width: 80px;
    height: 80px;
  }
  .loader:after {
    content: ' ';
    display: block;
    width: 25rem;
    height: 25rem;
    margin: 2rem;
    border-radius: 50%;
    border: 0.1rem solid var(--green);
    border-color: var(--green) transparent var(--green) transparent;
    animation: loader 1.2s linear infinite;
  }
  @keyframes loader {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }