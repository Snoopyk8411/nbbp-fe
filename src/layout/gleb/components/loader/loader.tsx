import loaderStyles from './loader.module.css';

export const Loader = (): JSX.Element => {
  return <div className={loaderStyles.loader}></div>;
};
