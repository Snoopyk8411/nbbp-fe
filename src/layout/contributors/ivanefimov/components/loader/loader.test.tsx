import renderer from 'react-test-renderer';

import { Loader } from './loader';

describe('Loader component', () => {
  it('should get valid snapshot', () => {
    const component = renderer.create(<Loader />);
    expect(component).toMatchSnapshot();
  });
});
