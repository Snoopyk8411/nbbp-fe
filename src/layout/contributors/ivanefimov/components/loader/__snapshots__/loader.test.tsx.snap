// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should get valid snapshot 1`] = `
<div
  className="loader"
/>
`;
